compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -160,7 +156,11 @@ public void perform(SyntaxNodeAnalysisContext ctx) {
                         validateRelations(field, entity, entity);
                     }
                 }
-
+                if (this.entities.size() > 0) {
```

Comment:
```suggestion
                if (!this.entities.isEmpty()) {
```
This can be moved into `validateEntityInMultipleSubModule` function

Reply:
Moved now